toolchain/sem_ir/file.h

```diff
@@ -173,15 +178,22 @@ class File : public Printable<File> {
     return id;
   }
 
+  // Adds a node block with the given content, returning an ID to reference it.
+  auto AddNodeBlock(llvm::ArrayRef<NodeId> content) -> NodeBlockId {
+    NodeBlockId id(node_blocks_.size());
+    node_blocks_.push_back({});
+    node_blocks_.back().assign(content.begin(), content.end());
```

Comment:
`node_blocks_.append(SmallVector<NodeId>(content.begin(), content.end()))` would create and initialize a temporary `SmallVector<NodeId>` and end up performing two copies of the content rather than one. We could use `emplace_back` to do this in one line, but ... this is being removed again in #3224 anyway so that'd just be churn.

Reply:
I'm not suggesting to create a `SmallVector`, just:

`node_blocks_.back().append(content.begin(), content.end());`

For reference, the `assign` implementation seems to delegate directly to `append`:

template <typename ItTy, typename = EnableIfConvertibleToInputIterator<ItTy>>
  void [assign](https://llvm.org/doxygen/classllvm_1_1SmallVectorImpl.html#aaf8c4a3f62d6231ae323b77e21303e1c)([ItTy](https://llvm.org/doxygen/classItTy.html) in_start, [ItTy](https://llvm.org/doxygen/classItTy.html) in_end) {
    this->[assertSafeToReferenceAfterClear](https://llvm.org/doxygen/classllvm_1_1SmallVectorTemplateCommon.html#a709b8b7b7dd6a5ea1f9e8255b1dd5b13)(in_start, in_end);
    [clear](https://llvm.org/doxygen/classllvm_1_1SmallVectorImpl.html#aac0ea55010b7b1a301e65a0baea057aa)();
    [append](https://llvm.org/doxygen/classllvm_1_1SmallVectorImpl.html#a7efd1f0c1206d95e4fe01a9b49a57b82)(in_start, in_end);
  }